Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1588,11 +1588,18 @@ class BrowserViewController: UIViewController {
     }
     
     func clearHistoryAndOpenNewTab() {
-        History.deleteAll { [weak self] in
-            guard let self = self else { return }
-            
-            self.tabManager.clearTabHistory() {
-                self.openBlankNewTab(attemptLocationFieldFocus: true, isPrivate: false)
+        // When PB Only mode is enabled
+        // All private tabs closed and a new private tab is created
+        if Preferences.Privacy.privateBrowsingOnly.value {
+            tabManager.removeAll()
+            openBlankNewTab(attemptLocationFieldFocus: true, isPrivate: true)
+        } else {
+            History.deleteAll { [weak self] in
+                guard let self = self else { return }
+                
+                self.tabManager.clearTabHistory() {
+                    self.openBlankNewTab(attemptLocationFieldFocus: true, isPrivate: false)
```

Comment:
If you are currently in private browsing mode here shouldn't we open a new private tab?

Reply:
I am not sure. But I believe this shortcut should open a new tab in normal mode after clearing the history. This is technically clearing the tabs in private mode with this way.